snooty/main.py

```diff
@@ -171,93 +203,132 @@ def _config_db(self) -> str:
             assert isinstance(db_name, str)
             return db_name
 
-    def on_update(
+    def on_update_metadata(
         self,
         prefix: List[str],
         build_identifiers: BuildIdentifierSet,
-        page_id: FileId,
-        page: Page,
+        field: Dict[str, SerializableType],
     ) -> None:
-        super().on_update(prefix, build_identifiers, page_id, page)
-
-        uploadable_assets = [
-            asset for asset in page.static_assets if asset.can_upload()
-        ]
-
-        fully_qualified_pageid = "/".join(prefix + [page_id.without_known_suffix])
+        property_name = "/".join(prefix)
```

Comment:
I would change this to `property_name_with_prefix` or something similarly contextual - at a glance on line 216 I assumed we removed prefixing the `page_id` and just had the property name ala `docs` or `drivers` until I double checked this line 😅